homeassistant/components/dweet.py

```diff
@@ -0,0 +1,68 @@
+"""
+A component which allows you to send data to Dweet.io.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/dweet/
+"""
+import logging
+from datetime import timedelta
+
+from homeassistant.const import EVENT_STATE_CHANGED, STATE_UNKNOWN
+from homeassistant.helpers import state as state_helper
+from homeassistant.helpers import validate_config
+from homeassistant.util import Throttle
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "dweet"
+DEPENDENCIES = []
+
+REQUIREMENTS = ['dweepy==0.2.0']
+
+CONF_WHITELIST = 'whitelist'
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=5)
+
+
+# pylint: disable=too-many-locals
+def setup(hass, config):
+    """Setup the Dweet.io component."""
+    if not validate_config(config,
+                           {DOMAIN: ['name', CONF_WHITELIST]},
+                           _LOGGER):
+        return False
+
+    conf = config[DOMAIN]
+    name = conf['name']
+    whitelist = conf.get(CONF_WHITELIST, [])
+    json_body = {}
+
+    def dweet_event_listener(event):
+        """Listen for new messages on the bus and sends them to Dweet.io."""
+        state = event.data.get('new_state')
+        if state is None or state.state in (STATE_UNKNOWN, '') \
+                or state.entity_id not in whitelist:
+            return
+
+        try:
+            _state = state_helper.state_as_number(state)
+        except ValueError:
+            _state = state.state
+
+        json_body[state.object_id] = _state
+
+        send_data(name, json_body)
+
+    hass.bus.listen(EVENT_STATE_CHANGED, dweet_event_listener)
+
+    return True
+
+
+@Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
Are they not accepting a couple of changes per second per 'thing' or in general? Right now after a publish of 1 state, you would not publish anything for 5 seconds. Since platforms are updated at once, this means that if you want to export two entities from 1 platform, only 1 state will make it out .